packages/graphql/src/services/accounts.ts

```diff
@@ -150,69 +168,75 @@ export async function sendMagicLink(
   { usernameOrEmail }: SendMagicLinkArgs,
   ctx: Context
 ) {
-  const participantWithUsername = await ctx.prisma.participant.findUnique({
-    where: { username: usernameOrEmail.trim() },
-  })
-  const participantWithEmail = await ctx.prisma.participant.findUnique({
-    where: { email: usernameOrEmail.trim().toLowerCase() },
-  })
+  const trimmedUsernameOrEmail = usernameOrEmail.trim()
 
-  const participant = participantWithUsername || participantWithEmail
+  // the returned count can never be more than one, as the username cannot be a valid email (and vice versa)
+  const participantWithUsername = await ctx.prisma.participant.findMany({
+    where: {
+      OR: [
+        { username: trimmedUsernameOrEmail },
+        { email: trimmedUsernameOrEmail.toLowerCase() },
+      ],
+    },
+  })
 
-  // participant does not exist, return early
-  if (!participant) return true
+  if (participantWithUsername.length === 0) return true
 
-  const magicLinkToken = uuidv4()
-  const magicLinkExpiration = dayjs().add(15, 'minute').toDate()
+  const participantData = participantWithUsername[0]
 
-  // set loginToken and loginTokenExpiresAt
-  await ctx.prisma.participant.update({
-    where: { id: participant.id },
-    data: {
-      magicLinkToken: magicLinkToken,
-      magicLinkExpiresAt: magicLinkExpiration,
+  const magicLinkJWT = JWT.sign(
+    {
+      sub: participantData.id,
+      role: UserRole.PARTICIPANT,
+      scope: UserLoginScope.OTP,
     },
-  })
+    process.env.APP_SECRET as string,
+    {
+      algorithm: 'HS256',
+      expiresIn: '30 minutes',
+    }
+  )
 
-  // TODO: send email to user with magic link
-  const magicLink = `${process.env.APP_STUDENT_SUBDOMAIN}/magicLogin?token=${magicLinkToken}&username=${participant.username}`
+  const magicLink = `${process.env.APP_STUDENT_SUBDOMAIN}/magicLogin?token=${magicLinkJWT}`
   console.log(magicLink)
 
+  // // TODO: send email to user with magic link
+  // nodemailer
+  //   .transport
+  //   // TODO smtp params
+  //   ()
+
   return true
 }
 
 export async function loginParticipantMagicLink(
-  { username, token }: { username: string; token: string },
+  { token }: { token: string },
   ctx: Context
 ) {
-  if (!username || !token) return null
+  //
+  const tokenData = JWT.verify(token, process.env.APP_SECRET as string) as {
```

Comment:
need to check if it throws an error when it is expired -> if yes, need to try/catch and return false to show "expired message"